osu.Game/Scoring/ScoreManager.cs

```diff
@@ -56,12 +56,44 @@ protected override ScoreInfo CreateModel(ArchiveReader archive)
             }
         }
 
+        public IEnumerable<ScoreInfo> GetScoresForBeatmapSet(int beatmapSetID, int rulesetID)
+        {
+            var scores = GetAllUsableScoresEnumerable()
+                .Where(s => !s.DeletePending && s.Beatmap.BeatmapSetInfoID == beatmapSetID && s.RulesetID == rulesetID);
```

Comment:
What's the point of extracting `GetAllUsableScoresEnumerable` (which kind of lies with its name) if you're checking `DeletePending` the second time in the `Where` anyways? Sure, it'll probably be optimised by EF, but it *is* kind of sloppy.